src/main/java/uk/gov/justice/hmpps/prison/service/AgencyPrisonerPayProfileService.kt

```diff
@@ -12,10 +13,17 @@ class AgencyPrisonerPayProfileService(
   private val agencyPrisonerPayProfileRepository: AgencyPrisonerPayProfileRepository,
 ) {
   fun getAgencyPrisonerPayProfile(agencyId: String): AgencyPrisonerPayProfile {
-    val result = agencyPrisonerPayProfileRepository
-      .findAgencyPrisonerPayProfileByAgyLocIdEqualsAndEndDateIsNullAndStartDateIsLessThanEqual(agencyId, LocalDate.now())
-      .orElseThrow(EntityNotFoundException.withId(agencyId))
-    return entityToModel(result)
+    val today = LocalDate.now()
+    val agencyProfilesList = agencyPrisonerPayProfileRepository
+      .findAgencyPrisonerPayProfileByAgyLocId(agencyId)
+      .filter { agy -> !agy.startDate.isAfter(today) && (agy.endDate == null || !agy.endDate.isBefore(today)) }
+
+    if (agencyProfilesList.isEmpty()) {
+      log.error("No AGY_PRISONER_PAY_PROFILES row is active for agency {} on {}", agencyId, today)
```

Comment:
don't think this should be `error`.  For the private prisons for example there are no active prisoner pay profiles.